File: src/main/java/flink/streaming/WordCount/WordCount.java
package flink.streaming.WordCount;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.api.java.utils.ParameterTool;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.util.Collector;


public class WordCount {
    public static void main(String[] args) throws Exception {
        final ParameterTool para = ParameterTool.fromArgs(args);
        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.getConfig().setGlobalJobParameters(para);

        DataStream<String> text;
        if (para.has("input")) {
            text = env.readTextFile(para.get("input"));
        } else {
            System.out.println("Executing WordCount example with default input data set.");
            System.out.println("Use --input to specify file input.");
            text = env.fromElements(WordCountData.WORDS);
        }

        DataStream<Tuple2<String, Long>> counter = text
                .flatMap(new Tokenizer())
                .keyBy(0)
                .sum(1);

        counter.print();
        env.execute();
    }

    public static class Tokenizer implements FlatMapFunction<String, Tuple2<String, Long>> {

        @Override
        public void flatMap(String s, Collector<Tuple2<String, Long>> collector) throws Exception {
            String [] tokens = s.toLowerCase().split("\\W+");
            for (String token : tokens)
                collector.collect(new Tuple2<>(token,1L));
        }
    }
}

File: src/main/scala/wordCount.scala
import org.apache.flink.api.java.utils.ParameterTool
import org.apache.flink.streaming.api.scala._
import org.apache.flink.streaming.api.windowing.time.Time

object socketWindowWordCount {
  case class WordWithCount(word: String, count: Long)
  def main(args: Array[String]): Unit = {
    var hostname = "localhost"
    var port = 9090

    try {
      val params = ParameterTool.fromArgs(args)
      hostname = if (params.has("hostname")) params.get("hostname") else "localhost"
      port = if (params.has("port")) params.getInt("post") else 9090
    } catch {
      case e: Exception => {
        System.err.print("No port specified. Please run 'SocketWindowWordCount " +
          "--hostname <hostname> --port <port>', where hostname (localhost by default) and port " +
          "is the address of the text server")
        System.err.println("To start a simple text server, run 'netcat -l <port>' " +
          "and type the input text into the command line")
        return
      }
    }

    val env : StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    val text = env.socketTextStream("localhost", port,'\n')
    val windowedCounts = text
      .flatMap {w => w.split("\\s")}
      .map{w => WordWithCount(w,1)}
      .keyBy("word")
      .timeWindow(Time.seconds(5), Time.seconds(1))
      .sum("count")
    windowedCounts.print().setParallelism(1)
    env.execute("Socket Window WordCount")
  }
}
